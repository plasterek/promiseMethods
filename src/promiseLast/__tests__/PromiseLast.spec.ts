import { promiseLast } from "../PromiseLast.service";

describe("promiseLast function", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  const success: string = "resolved";
  const succesDelayed: string = "resolvedWithDelay";
  const failure: string = "rejected";
  const resolvedPromise: Promise<string> = Promise.resolve(success);
  const resolvedPromise2: Promise<string> = new Promise((resolve) => setTimeout(() => resolve(succesDelayed), 200));
  const rejectedPromise: Promise<string> = new Promise((reject) => setTimeout(() => reject(failure), 300));

  describe("When all Promises in array are resolved", () => {
    it("It returns Promise with result that was last to resolve", async () => {
      //given
      const promiseArray: Promise<string>[] = [resolvedPromise2, resolvedPromise];
      //when
      const results: string | [] | undefined = await promiseLast(promiseArray);
      //then
      expect.assertions(1);
      expect(results).toBe(succesDelayed);
    });
  });

  describe("When any promise rejects in given array", () => {
    it("It will return rejected promise after all promises in given array are resolved", async () => {
      //given
      const promiseArray: Promise<string>[] = [resolvedPromise2, rejectedPromise];
      //when
      const result: string | [] | undefined = await promiseLast(promiseArray);

      //then
      expect.assertions(1);
      expect(result).toBe(failure);
    });
  });

  describe("When given array is empty", () => {
    it("It returns Promise with an empty array", async () => {
      //when
      const results: string | [] | undefined = await promiseLast([]);
      //then
      expect.assertions(1);
      expect(results).toMatchObject([]);
    });
  });
});
