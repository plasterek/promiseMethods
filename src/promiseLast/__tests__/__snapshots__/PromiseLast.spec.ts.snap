// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promiseLast function When all Promises in array are resolved It returns Promise with result that took most time to resolve 1`] = `"resolvedWithDelay"`;

exports[`promiseLast function When any promise rejects in given array It will return rejected promise after all promises in given array are resolved 1`] = `"rejected"`;

exports[`promiseLast function When given array is empty It returns Promise with an empty array 1`] = `[]`;
